### PR TITLE
python_site_packages_path can be specified in the build section

### PR DESCRIPTION
**Pull request by jjhelmus**
### Description

This is a implementation of CEP 17 in https://github.com/conda/ceps/pull/90

The location of the site-packages directory can be specified in `python_site_packages_path` in the build section of a recipe. To avoid unnecessarily increasing the size of repodata this metadata should only be specified when the package name is `python` and when the path is not the default for CPython. These suggestions are **not** checked or enforced by conda-build.
 
### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda-build/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by jjhelmus** [comment]
I created a few sample packages that use specify this metadata field for testing with downstream tools. There are available on anaconda.org in the `jjhelmus/label/sp_path` channel. 

There are three noarch packages with the following paths specified
* python 3.99.99 : `lib/python3.99t/site-packages` - A "free-threading" like path
* python 3.99.90 : `sample_path/from_the_file/index_json` - Something that stands out
* python 3.99.50 : `../../../../../invalid_path` : Should be rejected by tooling as being invalid

**Comment by jjhelmus** [comment]
@conda/builds-tools This PR is ready for a review.

File: news/5502-python_site_packages_path-can-be-specified-in-the-build-section [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* The location of the site-packages directory can be specified in `python_site_packages_path` in the build section of a recipe (#5502).
```
Comment (beeankha) [inline news/5502-python_site_packages_path-can-be-specified-in-the-build-section:3]:
```suggestion
* Add option for the location of the site-packages directory to be specified via `python_site_packages_path` in the build section of a recipe for any packages named `python`. (#5502)
```

Comment (jezdez) [reply news/5502-python_site_packages_path-can-be-specified-in-the-build-section:3]: Let's mention CEP-17 as well, like in https://github.com/conda/conda/pull/14256/files#diff-34b92149e59e8ccca3335eebd8e8be442c5683b59ead38a89f9a47cd460a3aaeR3

**Review by jezdez — DISMISSED** [review]

File: news/5502-python_site_packages_path-can-be-specified-in-the-build-section (hunk 1 above)
Comment (jezdez) [reply news/5502-python_site_packages_path-can-be-specified-in-the-build-section:3]: Like so?

```suggestion
* Add  support for CEP-17 that allows specifying the location of the site-packages directory with the `python_site_packages_path` build option for any packages named `python`. (#5502)
```

**Review by beeankha — DISMISSED** [review]

**Review by beeankha — APPROVED** [review]

**Comment by wolfv** [comment]
FYI we have a implementation in rattler build as well (will be released today):

https://github.com/prefix-dev/rattler-build/pull/1161

**Pull request merged**